client/dialogs/action-creators.ts

```diff
@@ -1,11 +1,26 @@
 import React from 'react'
+import { ThunkAction } from '../dispatch-registry'
+import { isStarcraftHealthy } from '../starcraft/is-starcraft-healthy'
 import { CloseDialog, OpenDialog } from './actions'
 import { DialogPayload, DialogType } from './dialog-type'
 
-export function openDialog(payload: DialogPayload): OpenDialog {
-  return {
-    type: '@dialogs/open',
-    payload,
+export function openDialog(payload: DialogPayload): ThunkAction {
+  return (dispatch, getState) => {
+    const { starcraft } = getState()
+
+    if (payload.type === DialogType.Settings && !isStarcraftHealthy({ starcraft })) {
```

Comment:
This is not a good change imo. This action creator should never be opening *different* dialogs than what you asked it to, and doing this here means we will almost certainly end up building a ton of special behavior into this function in the future so, please don't

Reply:
A cleaner solution is to write an action creator in the settings package to open the correct dialog, and use that wherever we open the settings. (I also generally think this approach is unnecessary now since we only support a single StarCraft version and the settings available are therefore always the same, but that's probably a larger change)

Reply:
Another alternative (that would be more similar to what was in the code previously) would be to make these 2 dialogs part of the same dialog, and just pick which one to render based on state at the time. This is probably also a larger change though, since you need to track when people specifically request to change the path as well